server/src/main/java/io/spine/examples/chatspn/server/chat/ChatCardProjection.java

```diff
@@ -27,51 +27,78 @@
 package io.spine.examples.chatspn.server.chat;
 
 import io.spine.core.Subscribe;
-import io.spine.examples.chatspn.ChatId;
-import io.spine.examples.chatspn.chat.ChatPreview;
-import io.spine.examples.chatspn.chat.GroupChatView;
-import io.spine.examples.chatspn.chat.MessagePreview;
-import io.spine.examples.chatspn.chat.PersonalChatView;
+import io.spine.examples.chatspn.ChatCardId;
+import io.spine.examples.chatspn.chat.ChatCard;
 import io.spine.examples.chatspn.chat.event.ChatMarkedAsDeleted;
 import io.spine.examples.chatspn.chat.event.GroupChatCreated;
+import io.spine.examples.chatspn.chat.event.MembersAdded;
+import io.spine.examples.chatspn.chat.event.MembersRemoved;
 import io.spine.examples.chatspn.chat.event.PersonalChatCreated;
+import io.spine.examples.chatspn.chat.event.UserLeftChat;
+import io.spine.examples.chatspn.message.MessageView;
 import io.spine.examples.chatspn.message.event.MessageContentUpdated;
 import io.spine.examples.chatspn.message.event.MessageMarkedAsDeleted;
 import io.spine.examples.chatspn.message.event.MessagePosted;
 import io.spine.server.projection.Projection;
 
+import static io.spine.examples.chatspn.chat.Chat.ChatType.CT_GROUP;
+import static io.spine.examples.chatspn.chat.Chat.ChatType.CT_PERSONAL;
+
 /**
- * Single chat preview.
+ * {@link ChatCard} projection.
  */
-public final class ChatPreviewProjection
-        extends Projection<ChatId, ChatPreview, ChatPreview.Builder> {
+public final class ChatCardProjection
+        extends Projection<ChatCardId, ChatCard, ChatCard.Builder> {
 
     @Subscribe
     void on(PersonalChatCreated e) {
-        var view = PersonalChatView
-                .newBuilder()
-                .setCreator(e.getCreator())
-                .setMember(e.getMember())
-                .vBuild();
-        builder().setId(e.getId())
-                 .setPersonalChat(view);
+        var viewer = builder().getCardId()
+                              .getUser();
+        var chat = builder().getCardId()
+                            .getChat();
+        String chatName;
+        if (e.getCreator()
+             .equals(viewer)) {
+            chatName = e.getMemberName();
```

Comment:
It would be nice to describe this (`if / else`) idea in the Proto description for this field.